tokio-trace-tower-http/src/lib.rs

```diff
@@ -100,19 +104,10 @@ where
     type Item = InstrumentedHttpService<T::Item>;
     type Error = T::Error;
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        if let Some(span) = &mut self.span {
-            let span2 = span.clone();
-            let inner = &mut self.inner;
-            span.enter(move || {
-                inner
-                    .poll()
-                    .map(|ready| ready.map(|svc| InstrumentedHttpService::with_span(svc, span2)))
-            })
-        } else {
-            self.inner
-                .poll()
-                .map(|ready| ready.map(|svc| InstrumentedHttpService::new(svc)))
-        }
+        let span = self.span.clone();
```

Comment:
This will clone the span _every_ time we poll the `InstrumentedMakeService`, but we only need it if the inner service returns `Ok(Async::Ready(...))`. This may not be a _huge_ performance hit, but it would probably be better to avoid it.

Since the `map` here is over a `Result` rather than a future, we can borrow `self.span` from the calling scope. I think we can just move `self.span.clone` into the closure. We _may_ need to close over an explicit ref if the borrow-checker doesn't like that?